rust/worker/src/blockstore/arrow/sparse_index.rs

```diff
@@ -99,6 +99,47 @@ impl SparseIndex {
         reverse.insert(block_id, SparseIndexDelimiter::Start);
     }
 
+    pub(super) fn get_all_target_block_ids(
+        &self,
+        search_keys: &mut Vec<CompositeKey>,
```

Comment:
hmm if the keyset is large is making these composite keys expensive?

Reply:
discussed offline to leave as is. The correct fix to remove composite key from sparse index abstraction